Qast/Managers/FirebaseManager.swift

```diff
@@ -8,24 +8,40 @@
 
 import FirebaseFirestore
 import CoreLocation.CLLocation
+import GeoFire
 
 struct FirebaseManager: SoundZoneAPI {
 
     private let datastore = Firestore.firestore()
     
     func soundZones(nearby location: CLLocationCoordinate2D, distance: Double = 30, _ completion: @escaping SoundZoneCompletionBlock) {
+        guard let queries = GFGeoHashQuery.queries(forLocation: location, radius: distance) else {
+            return completion([], NSError())
+        }
         
-        let bounds = FirebaseManager.boundingPoints(for: location, distance: distance)
-        
-        datastore.collection("sound_zones")
-            .whereField("center", isGreaterThan: bounds.lesserPoint)
-            .whereField("center", isLessThan: bounds.greaterPoint)
-            .getDocuments { (snapshot, error) in
-                if let error = error {
-                    completion([], error)
-                } else if let snapshot = snapshot {
-                    completion(snapshot.documents.compactMap({ SoundZone(dictionary: $0.data()) }), nil)
+        for query in queries {
+            guard let query = query as? GFGeoHashQuery else {
+                continue
+            }
+            
+            convert(query).getDocuments { (snapshot, error) in
+                guard let documents = snapshot?.documents, error == nil else {
+                    return completion([], error ?? NSError())
                 }
+                completion(documents.compactMap { SoundZone(dictionary: $0.data()) }, nil)
+            }
+        }
+    }
+    
+    func create(_ soundZone: SoundZone, completion: @escaping (Error?) -> Void) {
+        guard let geohash = GFGeoHash(location: soundZone.center.location).geoHashValue  else {
```

Comment:
Do we know if there's a minimum size to the geohashes generated from the center coordinate?

May just be semantic, but the way my team at Ford uses the word "geohash" they mean a 500 sq. ft. zone. Hard limitation of the geohasher we use.

Just wondering.

Reply:
https://gis.stackexchange.com/questions/115280/what-is-the-precision-of-a-geohash